radixdlt-core/radixdlt/src/test/java/com/radixdlt/stateir/StateIRSerializationTest.java

```diff
@@ -0,0 +1,282 @@
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.stateir;
+
+import static com.radixdlt.atom.TxAction.CreateMutableToken;
+import static org.junit.Assert.assertEquals;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.radixdlt.application.system.FeeTable;
+import com.radixdlt.application.tokens.Amount;
+import com.radixdlt.application.tokens.state.TokenResource;
+import com.radixdlt.atom.MutableTokenDefinition;
+import com.radixdlt.atom.TxAction;
+import com.radixdlt.atom.TxBuilderException;
+import com.radixdlt.atom.TxnConstructionRequest;
+import com.radixdlt.crypto.ECKeyPair;
+import com.radixdlt.engine.RadixEngine;
+import com.radixdlt.engine.RadixEngineException;
+import com.radixdlt.hotstuff.LedgerProof;
+import com.radixdlt.identifiers.REAddr;
+import com.radixdlt.mempool.MempoolConfig;
+import com.radixdlt.modules.SingleNodeAndPeersDeterministicNetworkModule;
+import com.radixdlt.statecomputer.LedgerAndBFTProof;
+import com.radixdlt.statecomputer.checkpoint.MockedGenesisModule;
+import com.radixdlt.statecomputer.forks.Forks;
+import com.radixdlt.statecomputer.forks.ForksModule;
+import com.radixdlt.statecomputer.forks.RERulesConfig;
+import com.radixdlt.statecomputer.forks.RadixEngineForksLatestOnlyModule;
+import com.radixdlt.statecomputer.forks.modules.MainnetForksModule;
+import com.radixdlt.store.DatabaseLocation;
+import com.radixdlt.store.LastStoredProof;
+import com.radixdlt.store.berkeley.BerkeleyLedgerEntryStore;
+import com.radixdlt.utils.PrivateKeys;
+import com.radixdlt.utils.UInt256;
+import java.io.ByteArrayInputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.stream.IntStream;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public final class StateIRSerializationTest {
+  private static final ECKeyPair VALIDATOR_KEY = PrivateKeys.ofNumeric(1);
+
+  @Rule public TemporaryFolder folder = new TemporaryFolder();
+
+  @Inject private RadixEngine<LedgerAndBFTProof> radixEngine;
+
+  @Inject private BerkeleyLedgerEntryStore ledgerEntryStore;
+
+  // FIXME: Hack, need this in order to cause provider for genesis to be stored
+  @Inject @LastStoredProof private LedgerProof ledgerProof;
```

Comment:
Could elaborate a bit more on the comment? Seems it is not being used in the code and removing it doesn't cause a compilation error.

Reply:
I think the genesis just isn't getting initialized unless the @LastStoredProof creation is triggered (by injecting it). TBH I've just copied it over from another similar test :sweat_smile:

Reply:
Should we create a task to investigate and improve that?

Reply:
I think this might not be worth the effort for olympia.

Reply:
This will actually need to be fixed as part of us moving to an initialization model for Babylon; agreed it's not worth fixing for Olympia! :)